e2e_tests/tests/deploy/test_local.py

```diff
@@ -192,3 +197,79 @@ def test_agent_up_down() -> None:
     assert len(containers) == 0
 
     master_down([])
+
+
+@pytest.mark.parametrize("steps", [10])
+@pytest.mark.parametrize("num_agents", [3, 5])
+@pytest.mark.parametrize("should_disconnect", [False, True])
+@pytest.mark.det_deploy_local
+def test_stress_agents_reconnect(steps: int, num_agents: int, should_disconnect: bool) -> None:
+    master_host = "localhost"
+    master_port = "8080"
+    conf.MASTER_IP = master_host
+    conf.MASTER_PORT = master_port
+    master_up([])
+
+    # Start all agents.
+    agents_are_up = [True] * num_agents
+    for i in range(num_agents):
+        agent_up(["--agent-name", f"agent-{i}"])
+    time.sleep(3)
+
+    for _ in range(steps):
+        for agent_id, agent_is_up in enumerate(agents_are_up):
+            if random.choice([True, False]):  # Flip agents status randomly.
+                continue
+
+            if agent_is_up:
```

Comment:
I'd switch them up / down randomly regardless of their current state, instead of "toggling" from up to down and back. that'd be more chaotic.